static/admin/admin.js

```
@@ -152,6 +152,54 @@ var GuestPreview = createClass({
 })
 CMS.registerPreviewTemplate("guests", GuestPreview)
 
+var HotelPreview = createClass({
+    render: function() {
+        const entry = this.props.entry.toJS()
+        const data = entry.data
+        const bookingStatus = data.bookingStatus
+        const isOpen = bookingStatus === 'open'
+        const isClosed = bookingStatus === 'closed'
+        if (!isOpen) {
+            let message = "Online hotel booking is now closed. Please [contact us] if you would like assistance booking a room at our discounted rates."
+            if (!isClosed) {
```

This is a bit of a nitpick but can we please use an `isWaiting` variable in both places instead of this? It's not super clear what this is doing during a quick read, and if we ever added another state this could lead to weird behavior.

---

Right! That helps a lot when reading the code. In that case, how should we handle invalid states?

---

Honestly? Don't, or maybe throw an error. 

There's a time and place for graceful degradation, but if we somehow get into a situation where booking status isn't one of the 3 we expect, that's a problem we need to be aware of. Anything else we show users would be wrong or potentially misleading.